gen/flights/flightplan.py

```diff
@@ -1546,16 +1557,37 @@ def generate_cas(self, flight: Flight) -> CasFlightPlan:
 
         builder = WaypointBuilder(flight, self.coalition)
 
+        # 2021-08-02: patrol_speed will currently have no effect because
+        # CAS doesn't use OrbitAction. But all PatrollingFlightPlan are expected
+        # to have patrol_speed
+        is_helo = flight.unit_type.dcs_unit_type.helicopter
+        logging.debug(f"Is {flight.unit_type.name} helicopter? {is_helo}")
+        ingress_egress_altitude = (
+            self.doctrine.ingress_altitude if not is_helo else feet(3000)
```

Comment:
https://github.com/dcs-liberation/dcs_liberation/blob/9121cf7ecb10d815e37b063af80351eaa62e26e4/gen/flights/waypointbuilder.py#L213

There's a lot of cleanup needed for helicopters, as you're discovering. Quick fix is to make sure these two files agree. I'd go with the quick fix, since the IP is a bit complicated (it should actually be _weapon_ dependent, as well as platform dependent).